packages/azure-kusto-data/source/kustoTrustedEndpoints.ts

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { default as endpointsData } from "./wellKnownKustoEndpoints.json";
+
+function getStringTailLowerCase(val: string, tailLength: number) {
+    if (tailLength <= 0) {
+        return "";
+    }
+
+    if (tailLength >= val.length) {
+        return val.toLowerCase();
+    }
+
+    return val.substring(val.length - tailLength).toLowerCase();
+}
+
+export class MatchRule {
```

Comment:
In typescript you can use the "Parameter Properties" shorthand, and the docs have a different syntax:

```typescript
export class MatchRule {
    constructor(
        /**
         * The suffix which the candidate must end with in order to match.
         */
        public suffix: string,
        /**
         * Indicates whether the match must be exact (the candidate must
         * not have any prefix) or not.         
         */
        public exact: boolean
    ) {}
}
```